### DISCUSSION TITLE
key_id deprecated. What's the alternative?

### DISCUSSION THREAD

**Discussion by kramer65** [latchset/jwcrypto#329] [Announcements]
I'm getting a deprecation warning saying: "DeprecationWarning: Call to deprecated function (or staticmethod) key_id."

But I see no mention of an alternative. What can I do about this warning?

**Comment by simo5** [comment]
For keys that have a kid you simply look for the "kid" attribute:
```
>>> key = jwk.JWK.generate(kty='oct', size=256, kid="test")
>>> key["kid"]
'test'
```

if you do not know if a key has a kid use the usual safer .get method:
```
>>> key = jwk.JWK.generate(kty='oct', size=256)
>>> key.get("kid")
>>> 
```
